updater/lib/dependabot/job.rb

```diff
@@ -399,6 +400,12 @@ def register_experiments
       end
     end
 
+    sig { void }
+    def validate_job
+      raise "Either directory or directories must be provided" if source.directory.nil? && source.directories.nil?
```

Comment:
Would it help to add more logs about what was actually set? And this also might be simpler as the XOR operation

Reply:
>Would it help to add more logs about what was actually set?

Nah, we log the job first thing before using this class and can also look it up via chat-op.